packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -0,0 +1,549 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+import 'dart:collection' show Queue;
+
+import 'package:flutter/widgets.dart';
+import 'package:meta/meta.dart';
+import 'package:vector_math/vector_math_64.dart' show Vector3;
+
+import 'colors.dart';
+import 'constants.dart';
+import 'icon.dart';
+import 'icon_theme.dart';
+import 'icon_theme_data.dart';
+import 'ink_well.dart';
+import 'material.dart';
+import 'theme.dart';
+
+const double _kActiveMaxWidth = 168.0;
+const double _kInactiveMaxWidth = 96.0;
+
+/// Signature of the [BottomNavigationBar.onAction] callback.
+///
+/// The [animation] argument can be used to coordinate with the action's
+/// animation. The [index] argument specifies which action has been tapped.
+typedef void BottomNavigationBarCallback(Animation<double> animation, int index);
+
+/// Defines the layout and behavior of a [BottomNavigationBar].
+///
+/// See also:
+///
+///  * [BottomNavigationBar]
+///  * [DestinationLabel]
+///  * <https://material.google.com/components/bottom-navigation.html#bottom-navigation-specs>
+enum BottomNavigationBarType {
+  /// Bottom navigation labels with fixed width.
+  fixed,
+
+  /// Bottom navigation labels with animated widths and positions.
```

Comment:
The location and size of the BottomNavigationBar DestinationLabels animate larger when then are tapped.